planner/core/rule_column_pruning.go

```diff
@@ -84,10 +73,7 @@ func exprHasSetVarOrSleep(expr expression.Expression) bool {
 // If any expression has SetVar function or Sleep function, we do not prune it.
 func (p *LogicalProjection) PruneColumns(parentUsedCols []*expression.Column) error {
```

Comment:
It seems `PruneColumns` don't need to return `error`.

Reply:
There's a error check in Aggregation's `PruneColumns`. In line 113.